app/services/water_management_area_data_load_service.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+require "zip"
+
+class WaterManagementAreaDataLoadService < FloodRiskEngine::BaseService
+  def run
+    geo_json = read_areas_file
+    features = RGeo::GeoJSON.decode(geo_json)
+
+    ActiveRecord::Base.transaction do
+
+      features.each do |feature|
+        area_name = feature.properties["long_name"]
+
+        attributes = {
+          area_name:,
+          area_id: feature.properties["identifier"],
+          long_name: feature.properties["long_name"],
+          short_name: feature.properties["short_name"],
+          area: feature.geometry
+        }
+
+        area = FloodRiskEngine::WaterManagementArea.where(code: feature["code"]).first
+
+        if area.present?
+          area.update(attributes)
+          puts "Updated Water Management Area \"#{area.code}\" (#{area.area_name})" unless Rails.env.test? # rubocop:disable Rails/Output
+        else
+          area = FloodRiskEngine::WaterManagementArea.create(attributes.merge(code: feature["code"]))
+          puts "Created Water Management Area \"#{area.code}\" (#{area.area_name})" unless Rails.env.test? # rubocop:disable Rails/Output
+        end
+      end
+    end
+  end
+
+  private
+
+  def read_areas_file
+    Zip::File.open(zipfile_path) do |zip_file|
+      zip_file.glob(areas_filename)
```

Comment:
Keeping an archive and extracting it on the fly looks like an interesting idea. But wouldn't that potentially increase system load, especially if there's no caching involved?  This means every service request would entail unzipping the archive.

Reply:
This is a once-off Rake task to load the boundary data into the DB so I think it should be OK.

Reply:
Ok, in that case that shouldn't be an issue